instrumentation/jsonrpc4j-1.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jsonrpc4j/v1_3/JettyServer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jsonrpc4j.v1_3;
+
+import com.googlecode.jsonrpc4j.AnnotationsErrorResolver;
+import com.googlecode.jsonrpc4j.JsonRpcServer;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Random;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+
+@SuppressWarnings("WeakerAccess")
+public class JettyServer implements AutoCloseable {
+
+  public static final String DEFAULT_LOCAL_HOSTNAME = "127.0.0.1";
+
+  public static final String SERVLET = "someSunnyServlet";
+  private static final String PROTOCOL = "http";
+
+  private final Class<?> service;
+
+  private Server jetty;
+  private int port;
+
+  JettyServer(Class<?> service) {
+    this.service = service;
+  }
+
+  public String getCustomServerUrlString(String servletName) {
+    return PROTOCOL + "://" + DEFAULT_LOCAL_HOSTNAME + ":" + port + "/" + servletName;
+  }
+
+  public void startup() throws Exception {
+    port = 10000 + new Random().nextInt(30000);
```

Comment:
This is not a good way for finding a port to use, there is a chance that you'll end up picking a port that is already used by some other tests that runs in parallel with this test. Instead try calling `new Server(0);` so that os would pick you a port and use `port = ((ServerConnector) jetty.getConnectors()[0]).getLocalPort();` to get the port that was used.